src/math/ec_scalar.cpp

```diff
@@ -38,6 +38,14 @@ ec_scalar::ec_scalar(const ec_secret& secret)
 {
 }
 
+ec_scalar::ec_scalar(uint64_t value)
+  : valid_(true)
+{
+    secret_ = zero;
+    auto serial = make_unsafe_serializer(secret_.end() - 4);
+    serial.write_4_bytes_big_endian(value);
```

Comment:
Writing 4 bytes casted from an 8 byte value, as big endian, into the high order bytes of a 32 byte value appears completely arbitrary and unrelated to the behavior of an ec scalar.